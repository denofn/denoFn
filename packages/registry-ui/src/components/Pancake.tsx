import React from "react";
import { css } from "otion";

export const Pancake = React.memo(function Pancake({
  className,
  children,
  onClick,
  padding,
}: {
  children?: React.ReactNode;
  className?: string;
  padding?: string;
  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
}) {
  const clickHandler = onClick ?? function () {};
  const clickCss = onClick
    ? {
        cursor: "pointer",
      }
    : {};
  return (
    <div
      onClick={clickHandler}
      className={`${css({
        ...clickCss,
        backgroundColor: "white",
        boxSizing: "border-box",
        borderRadius: ".25rem",
        minHeight: "5rem",
        display: "flex",
        alignItems: "flex-start",
        flexDirection: "column",
        justifyContent: "space-between",
        padding: padding ?? ".75rem",
        boxShadow: "0 10px 15px -3px rgba(0,0,0,.1), 0 4px 6px -2px rgba(0,0,0,0.05)",
      })}${className ? ` ${className}` : ""}`}
    >
      {children}
    </div>
  );
});
