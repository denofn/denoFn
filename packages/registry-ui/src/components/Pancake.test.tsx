import { render, cleanup } from "@testing-library/react";
import { css } from "otion";
import React from "react";

import { Pancake } from "./Pancake";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Pancake />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders flex=false with display:block", () => {
  render(<Pancake flex={false} />);
  expect(document.getElementsByClassName(css({ display: "block" })).length).toBe(1);
});
