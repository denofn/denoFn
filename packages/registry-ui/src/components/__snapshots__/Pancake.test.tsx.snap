// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_zjik7 _5dh3r6 _qdeacm _2o6p8u _1qesdsb _1khn195 _sy4y6p _1x4v58x _1dvcxr3 _u68uts"
  />
</DocumentFragment>
`;
